test/QtDBMigrationTest.cpp

```diff
@@ -134,6 +139,23 @@ void QtDBMigrationTest::migrateVersion_data()
         << "test_migrate_1.json"
         << (QList<int>() << 0 << 1 << 0)
         << (QStringList() << "table1");
+
+}
+void QtDBMigrationTest::migrateErrors()
+{
+    QFETCH(QString, configPath);
+
+    QtDBMigration mig(configPath);
+    bool ok = mig.migrate();
+    QVERIFY(!ok);
+}
+void QtDBMigrationTest::migrateErrors_data()
+{
+    QTest::addColumn<QString>("configPath");
+    QTest::newRow("Invalid path form")
+            << "random stirng not path";
+    QTest::newRow("Valid but empty path")
+            << QStandardPaths::writableLocation(QStandardPaths::HomeLocation);
```

Comment:
I would prefert teporary files. This way, tests are going to pollute home directory, see [this](https://doc.qt.io/qt-5/qtemporaryfile.html).

Reply:
Changed to temp files.